src/main/java/org/broadinstitute/hellbender/utils/read/CigarBuilder.java

```diff
@@ -34,11 +36,19 @@
 
     private Section section = Section.LEFT_HARD_CLIP;
 
+    private final boolean removeDeletionsAtEnds;
+
     private int leadingDeletionBasesRemoved = 0;
     private int trailingDeletionBasesRemoved = 0;
     private int trailingDeletionBasesRemovedInMake = 0;
 
-    public CigarBuilder() { }
+    public CigarBuilder(final boolean removeDeletionsAtEnds) {
+        this.removeDeletionsAtEnds = removeDeletionsAtEnds;
```

Comment:
Add a run of tests asserting both true and false cases for this variable.

Reply:
This is the `testRetainDeletions` unit test.